include/rdm/param_data.h

```diff
@@ -0,0 +1,585 @@
+etcpal_error_t rdmpd_unpack_get_resp_identify_device(const RdmParamData *pd, bool *identify_state_out);
+etcpal_error_t rdmpd_pack_get_resp_identify_device(bool identify_state, RdmParamData *pd_out);
+etcpal_error_t rdmpd_unpack_set_identify_device(const RdmParamData *pd, bool *identify_out);
+etcpal_error_t rdmpd_pack_set_identify_device(bool identify, RdmParamData *pd_out);
+
+// Get SUPPORTED_PARAMETERS See Section 10.4.1 of [RDM]
+/*! The maximum number of 16-bit parameter IDs in the Get Supported Parameters Response PD. */
+#define RDMPD_MAX_SUPPORTED_PARAMETERS 115
+
+/*! A structure representing supported parameters. */
+typedef struct RdmPdSupportedParameters
+{
+  uint16_t parameters[RDMPD_MAX_SUPPORTED_PARAMETERS]; /*!< The array of supported PIDs. */
+  size_t length;                                       /*!< The number of supported PIDs in the array. */
+} RdmPdSupportedParameters;
+
+etcpal_error_t rdmpd_unpack_get_resp_supported_parameters(const RdmParamData *pd, RdmPdSupportedParameters *params_out);
+etcpal_error_t rdmpd_pack_get_resp_supported_parameters(const RdmPdSupportedParameters *params, RdmParamData *pd_out);
+
+// Get PARAMETER_DESCRIPTION See Section 10.4.2 of [RDM]
+etcpal_error_t rdmpd_unpack_get_parameter_description(const RdmParamData *pd, uint16_t *requested_pid_out);
+etcpal_error_t rdmpd_pack_get_parameter_description(uint16_t requested_pid, RdmParamData *pd_out);
+
+/*! Data type enumeration */
+typedef enum
+{
+  kRdmPdDtNotDefined = 0x00,
+  kRdmPdDtBitField = 0x01,
+  kRdmPdDtAscii = 0x02,
+  kRdmPdDtUnsignedByte = 0x03,
+  kRdmPdDtSignedByte = 0x04,
+  kRdmPdDtUnsignedWord = 0x05,
+  kRdmPdDtSignedWord = 0x06,
+  kRdmPdDtUnsignedDword = 0x07,
+  kRdmPdDtSignedDword = 0x08,
+  kRdmPdDtManuSpecificStart = 0x80,
+  kRdmPdDtManuSpecificEnd = 0xdf
+} rdmpd_data_type_t;
+
+/*! Command class enumeration (get, set, or both) */
+typedef enum
+{
+  kRdmPdCcGet = 0x01,
+  kRdmPdCcSet = 0x02,
+  kRdmPdCcGetSet = 0x03
+} rdmpd_param_desc_cc_t;
+
+typedef enum
+{
+  kRdmPdSuNone = 0x00,
+  kRdmPdSuCentigrade = 0x01,
+  kRdmPdSuVoltsDc = 0x02,
+  kRdmPdSuVoltsAcPeak = 0x03,
+  kRdmPdSuVoltsAcRms = 0x04,
+  kRdmPdSuAmpereDc = 0x05,
+  kRdmPdSuAmpereAcPeak = 0x06,
+  kRdmPdSuAmpereAcRms = 0x07,
+  kRdmPdSuHertz = 0x08,
+  kRdmPdSuOhm = 0x09,
+  kRdmPdSuWatt = 0x0a,
+  kRdmPdSuKilogram = 0x0b,
+  kRdmPdSuMeters = 0x0c,
+  kRdmPdSuMetersSquared = 0x0d,
+  kRdmPdSuMetersCubed = 0x0e,
+  kRdmPdSuKilogrammesPerMeterCubed = 0x0f,
+  kRdmPdSuMetersPerSecond = 0x10,
+  kRdmPdSuMetersPerSecondSquared = 0x11,
+  kRdmPdSuNewton = 0x12,
+  kRdmPdSuJoule = 0x13,
+  kRdmPdSuPascal = 0x14,
+  kRdmPdSuSecond = 0x15,
+  kRdmPdSuDegree = 0x16,
+  kRdmPdSuSteradian = 0x17,
+  kRdmPdSuCandela = 0x18,
+  kRdmPdSuLumen = 0x19,
+  kRdmPdSuLux = 0x1a,
+  kRdmPdSuIre = 0x1b,
+  kRdmPdSuByte = 0x1c,
+  kRdmPdSuManuSpecificStart = 0x80,
+  kRdmPdSuManuSpecificEnd = 0xdf
+} rdmpd_sensor_unit_t;
+
+typedef enum
+{
+  kRdmPdSuPfxNone = 0x00,
```

Comment:
Just a personal style nitpick - I think `Pfx` should be expanded to `Prefix` here so people don't get abbreviation fatigue from parsing three abbreviations in a row.

Reply:
Done.